lua/starfall/libs_sh/color.lua

```diff
@@ -155,6 +196,17 @@ function color_methods:hsvToRGB()
 	return wrap({ rgb.r, rgb.g, rgb.b, (rgb.a or 255) })
 end
 
+local bit_tohex = bit.tohex
+--- Returns a hexadecimal string representation of the color
+-- @return String hexadecimal color
+function color_methods:getHex()
+	if self[4] == 255 then
+		return bit_tohex(self[1], 2)..bit_tohex(self[2], 2)..bit_tohex(self[3], 2)
```

Comment:
- `string.format` will be more efficient here.
- Instead of checking if the alpha channel is equal to 255, it needs an optional parameter that controls whether the alpha channel should be included in the returned string. This way the function behavior is more consistent.
- Shouldn't it be named `toHex` since it converts the color **to** the hexadecimal representation?

Reply:
Yeah good idea, I though it would be nifty, but you're right about the consistency.
Now made it so that alpha channel is never included by default, unless the optional param is `true`, then it always is regardless of it's value.